Automattic/woocommerce-payments
# Add WooPay marketing opt-in for selected countries

Fixes #5400

#### Changes proposed in this Pull Request

Add a new `Opt-in for marketing emails` (this text is not final) checkbox on WooPay sign up form when the user selects a billing address country that requires single opt-in, this includes the GDPR countries and countries that require explicit marketing consent (China, Singapore).









#### Testing instructions





This PR needs to be tested alongside 2878-gh-Automattic/woocommerce-payments-server and 1567-gh-Automattic/woopay.

It should be tested with classic checkout and blocks checkout.

**Disabling marketing**

* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) or China/Singapore.
* Click `Save my information for faster checkouts`.
* There should be a new `Opt-in for marketing emails` checkbox checked by default.
* Uncheck the `Opt-in for marketing emails` checkbox.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be unchecked.

**Enabling marketing**

* Sign out on WooPay.
* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) or China/Singapore.
* Click `Save my information for faster checkouts`.
* Let the `Opt-in for marketing emails` checkbox checked.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be checked.

**Other countries**

* Sign out on WooPay.
* Add some product to cart on the merchant site.
* On checkout page, choose an email that doesn't exist on WooPay.
* Do not select one of the [GDPR countries](https://www.gdpradvisor.co.uk/gdpr-countries) nor China/Singapore.
* Click `Save my information for faster checkouts`.
* There should not be a `Opt-in for marketing emails` checkbox.
* Place Order.
* Sign in on WooPay with the user you just created.
* Access the Account page.
* The `Opt-in for marketing emails` checkbox on account page should be checked.

-------------------


**Post merge**
